site/content/faq/how-do-you-configure-zap-logging.md

```diff
@@ -39,16 +39,14 @@ Logging can be selectively enabled using a [Stand Alone script](/docs/desktop/ad
 running (the example below is a JavaScript script):
 
 ```JavaScript
-var Level = Java.type("org.apache.logging.log4j.Level")
-var LoggerContext = Java.type("org.apache.logging.log4j.core.LoggerContext")
-
-var context = LoggerContext.getContext()
-var config = context.getConfiguration()
+// https://logging.apache.org/log4j/log4j-2.4/faq.html#reconfig_level_from_code
+var Configurator = Java.type("org.apache.logging.log4j.core.config.Configurator");
+var Level = Java.type("org.apache.logging.log4j.Level"); 
 
 // The following will enable DEBUG logging for the API
-config.getLoggerConfig("org.zaproxy.zap.extension.api.API").setLevel(Level.DEBUG)
+Configurator.setLevel("org.zaproxy.zap.extension.api.API", Level.DEBUG);
 // The following will enable DEBUG logging for the Session Fixation scan rule
-config.getLoggerConfig("org.zaproxy.zap.extension.ascanrulesBeta.SessionFixationScanRule").setLevel(Level.DEBUG)
-
-context.updateLoggers()
+Configurator.setLevel("org.zaproxy.zap.extension.ascanrulesBeta.SessionFixationScanRule", Level.DEBUG);
+// The following will enable DEBUG logging for the spider add-on
+Configurator.setLevel("org.zaproxy.addon.spider", Level.DEBUG);
```

Comment:
I'd suggest doing it in one go, e.g.:
```JavaScript
Configurator.setLevel({
  // The following will enable DEBUG logging for the API
  "org.zaproxy.zap.extension.api.API" : Level.DEBUG,
  // The following will enable DEBUG logging for the Session Fixation scan rule
  "org.zaproxy.zap.extension.ascanrulesBeta.SessionFixationScanRule" : Level.DEBUG,
  // The following will enable DEBUG logging for the spider add-on
  "org.zaproxy.addon.spider" : Level.DEBUG,
});
```
Less verbose and more efficient as it only updates the loggers once.